File: problem_02/problem_02.js
const top_number = 4_000_000;
let solution = 2;

function sum_even_fibonacci_numbers() {
    let a = 1;
    let b = 2;
    let fibonacci;
    let sum = b;

    while (a + b < top_number) {
        fibonacci = a + b;
        if (fibonacci % 2 === 0) {
            sum += fibonacci;
        }
        a = b;
        b = fibonacci;
    }
    return sum;
}

console.log("The sum of the even Fibonacci numbers is: ", sum_even_fibonacci_numbers());

File: problem_02/Problem_02.java
public class Problem_02 {
    public int sum_even_fibonacci(int top_number) {

        // Initialize variables
        int a = 1;
        int b = 2;
        int number;
        int sum = b;
        
        // Calculate the sum of the even Fibonacci numbers
        while (a + b < top_number) {
            number = a + b;
            if (number % 2 == 0) {
                sum += number;
            }            
            a = b;
            b = number;
        }
        return sum;
    }

    public static void main(String[] args) {
        // Create an instance of the class
        Problem_02 problem_02 = new Problem_02();

        // Initialize variables
        int top_number = 4_000_000;
        int solution = problem_02.sum_even_fibonacci(top_number);

        // Print the solution
        System.out.println("The sum of all the even Fibonacci numbers below " + top_number + " is " + solution);
   
    }
}